articles/ros2_threat_model.md

```diff
@@ -0,0 +1,3421 @@
+    <td class="danger">✘</td>
+    <td class="success">✓</td>
+    <td>
+      <ul>
+        <li>Long-range communication should always use a secure transport layer
+(WPA2 for WiFi for instance)</li>
+      </ul>
+    </td>
+    <td><a href="http://arxiv.org/abs/1504.04339">Bonaci, Tamara, Jeffrey
+Herron, Tariq Yusuf, Junjie Yan, Tadayoshi Kohno, and Howard Jay Chizeck. “To
+Make a Robot Secure: An Experimental Analysis of Cyber Security Threats Against
+Teleoperated Surgical Robots.” ArXiv:1504.04339 [Cs], April 16, 2015.</a></td>
+  </tr>
+
+  <tr>
+    <td>An attacker intercepts robot long-range communications (e.g. MitM)</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="success">✓</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="success">1</td>
+    <td class="warning">2</td>
+    <td class="success">1</td>
+    <td class="danger">3</td>
+    <td class="success">1</td>
+    <td>8</td>
+    <td class="success">✓</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="success">✓</td>
+    <td class="success">✓</td>
+    <td class="success">✓</td>
+    <td class="success">✓</td>
+    <td class="success">✓</td>
+    <td class="success">✓</td>
+    <td class="success">✓</td>
+    <td class="success">✓</td>
+    <td class="success">✓</td>
+    <td class="success">✓</td>
+    <td>
+      <ul>
+        <li>Long-range communication should always use a secure transport layer
+(WPA2 for WiFi for instance)</li>
+      </ul>
+    </td>
+    <td><a href="http://arxiv.org/abs/1504.04339">Bonaci, Tamara, Jeffrey
+Herron, Tariq Yusuf, Junjie Yan, Tadayoshi Kohno, and Howard Jay Chizeck. “To
+Make a Robot Secure: An Experimental Analysis of Cyber Security Threats Against
+Teleoperated Surgical Robots.” ArXiv:1504.04339 [Cs], April 16, 2015.</a></td>
+  </tr>
+
+  <tr>
+    <td>An attacker disrupts (e.g. jams) robot long-range communication
+channels.</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="success">✓</td>
+    <td class="danger">✘</td>
+    <td class="warning">2</td>
+    <td class="warning">2</td>
+    <td class="success">1</td>
+    <td class="success">1</td>
+    <td class="danger">3</td>
+    <td>9</td>
+    <td class="success">✓</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="success">✓</td>
+    <td class="success">✓</td>
+    <td class="success">✓</td>
+    <td class="success">✓</td>
+    <td class="warning">▲</td>
+    <td class="success">✓</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="success">✓</td>
+    <td class="danger">✘</td>
+    <td>
+      <ul>
+        <li>Multiple long-range communication transport layers should be used
+when possible (e.g. cellular and WiFi)</li>
```

Comment:
And the system should be designed to "fail closed." If the system isn't being commanded to do something or detects that it's lost communication, it might be a good time to halt.